src/App.js

```diff
@@ -1,26 +1,104 @@
 import React from 'react';
 import logo from './logo.svg';
 import './App.css';
+import Contacts from './contacts.json';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+class App extends React.Component {
+  state = {
+    listOfContacts: Contacts.slice(0, 5),
+  };
+
+  addRandomContact() {
+    const copyOfContactList = [...this.state.listOfContacts];
+
+    const addRandomContact =
+      Contacts[Math.floor(Math.random() * Contacts.length)];
+
+    copyOfContactList.push(addRandomContact);
+
+    this.setState({
+      listOfContacts: copyOfContactList,
+    });
+  }
```

Comment:
Because contacts that are pushed to the copyOfContactList that is set to state, are not spliced from Contacts, there can be duplicates on the page.